src/plugins/auto/docs/tests.md

```diff
@@ -49,4 +49,270 @@ Note: Google Test filter = *AutoReleaseHelperTest*cpuLoadFailure_accelerateorLoa
+
+```bash
+{
+   "AUTO": {
+            "ENABLE_STARTUP_FALLBACK": "YES"
+   }
+}
+```
+
+The retrieved property ``EXECUTION_DEVICE`` from AUTO will be CPU accelerator (``(CPU)``).
+
+```bash
+openvino/bin/intel64/Release$ ./benchark_app -m openvino/src/core/tests/models/ir/add_abc.xml -d AUTO -load_config ./config.json
+[Step 1/11] Parsing and validating input arguments
+[ INFO ] Parsing input parameters
+[Step 2/11] Loading OpenVINO Runtime
+[ INFO ] OpenVINO:
+[ INFO ] Build ................................. <OpenVINO version>-<Branch name>
+[ INFO ] 
+[ INFO ] Device info:
+[ INFO ] AUTO
+[ INFO ] Build ................................. <OpenVINO version>-<Branch name>
+[ INFO ] 
+[ INFO ] 
+...
+[ INFO ]   EXECUTION_DEVICES: (CPU)
+...
+[ INFO ] First inference took 0.65 ms
+...
+[ INFO ] Count:               169420 iterations
+[ INFO ] Duration:            120004.85 ms
+[ INFO ] Latency:
+[ INFO ]    Median:           2.76 ms
+[ INFO ]    Average:          2.78 ms
+[ INFO ]    Min:              0.51 ms
+[ INFO ]    Max:              8.39 ms
+[ INFO ] Throughput:          1411.78 FPS
+```
+
+Running benchmark_APP with disabling the property ``ENABLE_STARTUP_FALLBACK`` in JSON file ``config.json``.
+
+```bash
+{
+   "AUTO": {
+            "ENABLE_STARTUP_FALLBACK": "NO"
+   }
+}
+```
+
+The retrieved property ``EXECUTION_DEVICE`` from AUTO will be CPU accelerator (``GPU``).
+
+```bash
+openvino/bin/intel64/Release$ ./benchark_app -m openvino/src/core/tests/models/ir/add_abc.xml -d AUTO -load_config ./config.json
+[Step 1/11] Parsing and validating input arguments
+[ INFO ] Parsing input parameters
+[Step 2/11] Loading OpenVINO Runtime
+[ INFO ] OpenVINO:
+[ INFO ] Build ................................. <OpenVINO version>-<Branch name>
+[ INFO ] 
+[ INFO ] Device info:
+[ INFO ] AUTO
+[ INFO ] Build ................................. <OpenVINO version>-<Branch name>
+[ INFO ] 
+[ INFO ] 
+...
+[ INFO ]   EXECUTION_DEVICES: GPU
+...
+[ INFO ] First inference took 3.97 ms
+...
+[ INFO ] Count:               167560 iterations
+[ INFO ] Duration:            120003.96 ms
+[ INFO ] Latency:
+[ INFO ]    Median:           2.76 ms
+[ INFO ]    Average:          2.81 ms
+[ INFO ]    Min:              0.78 ms
+[ INFO ]    Max:              5.99 ms
+[ INFO ] Throughput:          1396.29 FPS
+```
+
+### Device selection fallback of the AUTO
+
+This section will show the fallback of device selection within the AUTO plugin if the device with high priority doesn't support the precision of the inputting model. For example, CPU supports both FP16 and FP32 precision model, while GNA doesn't FP32 precision. Although GNA has higher priority, AUTO plugin will ultimately choose the CPU plugin to load model with FP32 precision.  
+
+AUTO will select GNA if no other device is specified in the device candidate list.
+
+```bash
+openvino/bin/intel64/Release$ ./benchmark_app -m openvino/src/core/tests/models/ir/add_abc.xml -d AUTO:GNA -t 10
```

Comment:
Any reason we don't use GPU as example?

Reply:
GPU supports the model precision with both FP16 and FP32, which means selection fallback will not happen here if we specified `-d AUTO:GPU,CPU` for the model with precision FP32(add_abc.xml). GNA doesn't supports FP32 and we can see the selection fallback here when inputting `-d AUTO:GNA,CPU` for FP32 model.